src/preact/context.js

```diff
@@ -77,3 +85,35 @@ export function useAmpContext() {
   const AmpContext = getAmpContext();
   return useContext(AmpContext);
 }
+
+/**
+ * Whether the calling component should currently be in the loaded state.
+ *
+ * @param {!Loading|string} loadingProp
+ * @return {boolean}
+ */
+export function useLoad(loadingProp) {
```

Comment:
Can the `useLoad` hook automatically grab `loadingProp` from context so that it doesn't need to be passed?

Reply:
That's the key: a prop and a context mean slightly different things, so the math around them is the same:

1. Context means that this loading rule applies to a whole subtree.
2. A loading prop provides an "instruction" to the specific element.

The important principal here is that the lazy loading is always done per-component, never for a subtree. The eager loading just leaves the loading up to the browser. So it's an overlap of the context and a prop that give us the right loading instruction that should be used for this specific element.

To give an example: 
* The fact that we loaded carousel doesn't mean that we should load all children of the carousel. These are still scheduled individually.
* But, if we unload carousel, we have to unload all children.

Another thing to add: the context is fully hierarchical and optional in Preact and it always applies to the whole subtrees. So it doesn't fit very well a model of a per-component loading.

Reply:
Oh I see. So each `<Component` would be using normal react props as the `loadingProp` here, it just happens to have an overloaded name with the ContextProp (LoadingProp).

Reply:
Yes. Same type and very similar meaning. But also some differences. Really the best way to say it is "loading configuration for a subtree context" vs "loading configuration for the specific element".